### PR TITLE
[TLS] Replace PKCS #1 legacy methods

### PR DESCRIPTION
**Pull request by mtury**
This removes the PKCS legacy methods in pkcs1.py, thanks to an (almost?) undocumented `md5-sha1` hash function in openssl which reproduces the signature scheme in TLS <1.0. I just had to register it through the cryptography library `HashAlgorithm` interface. This is neat.

Other minor changes:

- Sort through default values for sign/verify and decrypt/encrypt methods
- Move one AC method to cert.py, and delete two others which had nothing to do in pkcs1.py
- Typos in cert.uts

### CODE REVIEW

**Comment by gpotter2** [comment]
Huh, openssl is really badly supported in windows, is there any way we avoid using it ?

File: scapy/layers/tls/crypto/pkcs1.py [hunk 1]
```diff
@@ -78,800 +61,87 @@ def pkcs_ilen(n):
...
-            fname = os.path.join(folder, fmtstr % i)
-            f = open(fname, "w")
-            s = a.output(fmt="PEM")
-            f.write(s)
-            f.close()
-            i += 1
-    except:
-        return None
-
-    r,w=popen2.popen2("c_rehash %s" % folder)
```
Comment (gpotter2) [inline scapy/layers/tls/crypto/pkcs1.py]: To be more efficiently useable, on windows, we're using a class with the external programs.
Could you integrate this with https://github.com/secdev/scapy/blob/master/scapy/arch/windows/__init__.py#L184 ?

Comment (mtury) [inline scapy/layers/tls/crypto/pkcs1.py]: I almost didn't pay attention to this part when I moved it. The function should be extremely rarely used and I don't think scapy should be used for this anyway. So I ended up removing it.

**Comment by mtury** [comment]
That was too good to be true: it seems that openssl does not ship by default with this `md5-sha1`. It's been in the code for a long time and I don't know yet why I had no trouble using it locally. I'll look into this later.

**Comment by mtury** [comment]
Most probably the Ubuntu and OSX openssl packages were modified from the OSes respective teams, and that's why my results on Debian differ from the ones computed by Travis.

Unfortunately there is no interface in the cryptography library even for raw RSA signatures, which I could have manually combined to a MD5 hash and a SHA1 hash. So I ended up restoring part of the legacy methods, which are called whenever the system's openssl does not know of a `md5-sha1` hash.

**Comment by mtury** [comment]
Please consider PR #542 in order to avoid any later conflicts.

**Pull request closed without merging**